terraform/cloud-platform-components/README.md

```diff
@@ -82,6 +83,114 @@ Metrics-server allows us to perform resource queries against the cluster. Comman
 ## Nginx-ingress
 A vital component in the cluster. The Nginx-ingress controller is a daemon, deployed as a Kubernetes Pod, that watches the apiserver's /ingresses endpoint for updates to the Ingress resource. Its job is to satisfy requests for Ingresses.
 
+## Open-Policy-Agent
+
+OPA is a lightweight general-purpose policy engine that can be co-located with our service. Services offload policy decisions to OPA by executing queries. OPA evaluates policies and data to produce query results (which are sent back to the client).  
+
+We utilise [opa-helm-chart](https://github.com/helm/charts/tree/master/stable/opa) to deploy opa onto the Cloud Platform. This helm chart installs OPA as a Kubernetes admission controller. In our case we are using validating admission controller. The helm Chart will automatically generate a CA and server certificate for the OPA. 
+
+In Kubernetes, Admission Controllers enforce semantic validation of objects during create, update, and delete operations. With OPA we can enforce custom policies on Kubernetes objects without recompiling or reconfiguring the Kubernetes API server. Please see the following documentation by opa on Kubernetes Admission Control 
+https://www.openpolicyagent.org/docs/kubernetes-admission-control.html
+
+To restrict the kinds of operations and resources that are subject to OPA policy checks, see the below example. Configured in templates/opa/values.yaml.tpl under admissionControllerRules:
+
+```yaml
+    admissionControllerRules:
+      - operations: ["CREATE", "UPDATE"]
+        apiGroups: ["extensions", ""]
+        apiVersions: ["v1beta1”, “v1”]
+        resources: ["ingresses", "namespaces"]
+```
+The admissionControllerRules the operations and resources that the webhook will validate. Intercept API requests when a ingress or a namespace is CREATED or UPDATED, so apiGroups and apiVersions are filled out accordingly (extensions/v1beta1 for ingresses, v1 for namespaces). We can use wildcards (*) for these fields as well.
+
+ ### kube-mgmt
+
+kube-mgmt manages instances of the Open Policy Agent on top of kubernetes. Use kube-mgmt to:
+
+    - Load policies into OPA via kubernetes.
```

Comment:
This renders as a code block. Can you please remove the indentation in order to change this to a list?

Reply:
Thanks, updated accordingly.